functions/go/set-annotation/main.go

```diff
@@ -58,51 +58,69 @@ func setAnnotation(spec setAnnotationSpec,
 
 //nolint
 func main() {
-	var plugin *plugin = &KustomizePlugin
-	tc, err := getDefaultConfig()
-	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
-	}
-
-	resmapFactory := newResMapFactory()
-
-	pluginHelpers := newPluginHelpers(resmapFactory)
 
 	resourceList := &framework.ResourceList{}
 	resourceList.FunctionConfig = map[string]interface{}{}
 
 	cmd := framework.Command(resourceList, func() error {
-		resMap, err := resmapFactory.NewResMapFromRNodeSlice(resourceList.Items)
+		err := run(resourceList)
 		if err != nil {
-			return err
-		}
-		annotations, err := getAnnotations(resourceList.FunctionConfig)
-		if err != nil {
-			return fmt.Errorf("failed to get data.specs field from function config: %w", err)
-		}
-		for _, a := range annotations.Annotations {
-			err := setAnnotation(a, resMap, tc, pluginHelpers, plugin)
-			if err != nil {
-				return fmt.Errorf("failed to add annotation [%s: %s]: %w",
-					a.AnnotationName, a.AnnotationValue, err)
+			resourceList.Result = &framework.Result{
+				Name: "set-annotation",
+				Items: []framework.Item{
+					{
+						Message:  err.Error(),
+						Severity: "error",
```

Comment:
I guess we can use this:
https://github.com/kubernetes-sigs/kustomize/blob/9a94c5ecd382dd4d9877f679fba7f91d9fb9645e/kyaml/fn/framework/types.go#L30